pkg/resources/cloudcontroller/digitalocean.go

```diff
@@ -0,0 +1,129 @@
+package cloudcontroller
+
+import (
+	"fmt"
+
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/resources"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+	"k8c.io/kubermatic/v2/pkg/resources/vpnsidecar"
+	appsv1 "k8s.io/api/apps/v1"
+	_ "k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
+	_ "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	DigitalOceanCCMDeploymentName = "digital-ocean-cloud-controller-manager"
+	DigitalOceanVersion           = "0.1.37"
+)
+
+func digitalOceanDeploymentCreator(data *resources.TemplateData) reconciling.NamedDeploymentCreatorGetter {
+	return func() (string, reconciling.DeploymentCreator) {
+		return DigitalOceanCCMDeploymentName, func(dep *appsv1.Deployment) (*appsv1.Deployment, error) {
+			dep.Name = DigitalOceanCCMDeploymentName
+			dep.Labels = resources.BaseAppLabels(DigitalOceanCCMDeploymentName, nil)
+			dep.Spec.Replicas = resources.Int32(1)
+
+			dep.Spec.Selector = &metav1.LabelSelector{
+				MatchLabels: resources.BaseAppLabels(DigitalOceanCCMDeploymentName, nil),
+			}
+
+			podLabels, err := data.GetPodTemplateLabels(DigitalOceanCCMDeploymentName, dep.Spec.Template.Spec.Volumes, nil)
+			if err != nil {
+				return nil, err
+			}
+
+			dep.Spec.Template.ObjectMeta = metav1.ObjectMeta{
+				Labels: podLabels,
+			}
+
+			dep.Spec.Template.Spec.DNSPolicy, dep.Spec.Template.Spec.DNSConfig, err =
+				resources.UserClusterDNSPolicyAndConfig(data)
+			if err != nil {
+				return nil, err
+			}
+
+			dep.Spec.Template.Spec.AutomountServiceAccountToken = pointer.Bool(false)
+
+			dep.Spec.Template.Spec.Volumes = append(getVolumes(data.IsKonnectivityEnabled()),
+				corev1.Volume{
+					Name: resources.CloudConfigConfigMapName,
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: resources.CloudConfigConfigMapName,
+							},
+						},
+					},
+				})
+
+			dep.Spec.Template.Spec.Containers = []corev1.Container{
+				{
+					Name:    ccmContainerName,
+					Image:   data.ImageRegistry(resources.RegistryMCR) + "/oss/kubernetes/digital-ocean-cloud-controller-manager:v" + DigitalOceanVersion,
```

Comment:
The image that you should use is `docker.io/digitalocean/digitalocean-cloud-controller-manager:v0.1.39`. Instead of `docker.io`, use `data.ImageRegistry(resources.RegistryDocker)`.